nix/default.nix

```diff
@@ -164,6 +142,73 @@ let
         '';
       });
 
+  bundlePursProject =
+    { name ? "${projectName}-bundle-" +
+        (if browserRuntime then "web" else "nodejs")
+    , entrypoint ? "index.js"
+    , htmlTemplate ? "index.html"
+    , main ? "Main"
+    , browserRuntime ? true
+    , webpackConfig ? "webpack.config.js"
+    , bundledModuleName ? "output.js"
+    , ...
+    }@args:
+    (buildPursProject (args // { withDevDeps = true; })).overrideAttrs
+      (oas: {
+        inherit name;
+        buildInputs = oas.buildInputs ++ [ nodejs ];
+        buildPhase = oas.buildPhase + ''
+          ${pkgs.lib.optionalString browserRuntime "export BROWSER_RUNTIME=1"}
+          chmod -R +rwx .
+          spago bundle-module --no-install --no-build -m "${main}" \
+            --to ${bundledModuleName}
+          cp $src/${entrypoint} .
+          cp $src/${htmlTemplate} .
+          cp $src/${webpackConfig} .
+          mkdir ./dist
+          webpack --mode=production -c ${webpackConfig} -o ./dist \
+            --entry ./${entrypoint}
+        '';
+        installPhase = ''
+          mkdir $out
+          mv dist $out
+        '';
+      });
+
+  buildPursDocsSearch =
+    { name ? "purescript-docs-search"
+    , ...
+    }:
+    let
+      docsUrl =
+        "https://github.com/purescript/purescript-docs-search/releases/download";
+      docsVersion = "v0.0.11";
```

Comment:
Consider using flake inputs for this and then `dependency inject` this like:

```
    purs-docs-search-app = {
      url  = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/docs-search-app.js";
      flake = false;
      };
    purs-docs-search = {
      url  = "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/purescript-docs-search";
      flake = false;
      };
```
But I understand the convenience of the current approach, so which ever you find nicer...

Reply:
Yes, that would be ideal